src/diffusers/models/vq_diffusion_attention.py

```diff
@@ -0,0 +1,246 @@
+from typing import Optional
```

Comment:
I think we don't really need a new file here as we should be able to pretty much integrate everything into the existing `attention.py`

Reply:
done!